scib/metrics/trajectory.py

```
@@ -60,44 +62,55 @@ def get_root(
         Column can contain empty entries, the dataset will be subset to the cells with scores.
+    :param batch_key: set to batch key if if you want to compute the trajectory metric by batch
     """
     # subset to cells for which pseudotime has been computed
     cell_subset = adata_pre.obs.index[adata_pre.obs[pseudotime_key].notnull()]
-    adata_pre_sub = adata_pre[cell_subset]
-    adata_post_sub = adata_post[cell_subset]
-
+    adata_pre_ti = adata_pre[cell_subset]
+    adata_post_ti = adata_post[cell_subset]
     try:
-        iroot, adata_post_sub2 = get_root(adata_pre_sub, adata_post_sub, label_key, pseudotime_key)
+        iroot, adata_post_ti2 = get_root(adata_pre_ti, adata_post_ti, label_key, pseudotime_key)
     except RootCellError:
         print('No root cell found, setting trajectory conservation metric to 0.')
         return 0  # failure to find root cell means no TI conservation
 
-    adata_post_sub2.uns['iroot'] = iroot
-
-    sc.tl.dpt(adata_post_sub2)  # stored in 'dpt_pseudotime'
-    adata_post_sub2.obs.loc[adata_post_sub2.obs['dpt_pseudotime'] > 1, 'dpt_pseudotime'] = 0
-    adata_post_sub.obs['dpt_pseudotime'] = 0
-    adata_post_sub.obs['dpt_pseudotime'] = adata_post_sub2.obs['dpt_pseudotime']
-    adata_post_sub.obs['dpt_pseudotime'].fillna(0, inplace=True)
-
-    pseudotime_before = adata_post_sub.obs['dpt_pseudotime']
-    pseudotime_after = adata_pre_sub.obs[pseudotime_key]
-    correlation = pseudotime_before.corr(pseudotime_after, 'spearman')
-
-    return (correlation + 1) / 2  # scaled
-
+    adata_post_ti2.uns['iroot'] = iroot
+
+    sc.tl.dpt(adata_post_ti2)  # stored in 'dpt_pseudotime'
+    adata_post_ti2.obs.loc[adata_post_ti2.obs['dpt_pseudotime'] > 1, 'dpt_pseudotime'] = 0
+    adata_post_ti.obs['dpt_pseudotime'] = 0
+    adata_post_ti.obs['dpt_pseudotime'] = adata_post_ti2.obs['dpt_pseudotime']
+    adata_post_ti.obs['dpt_pseudotime'].fillna(0, inplace=True)
+
+    adata_post_ti.obs['batch'] = adata_pre_ti.obs['batch']
```

Is this okay, @danielStrobl? Why should this not be stored in `adata_post_ti` already?

---

I'd suggest we assume that the batches should match before the function is called. Ideally this should be something that the integration method wrapper or the user take care of.
Alternatively I'd throw an error message if batches mismatch. We might risk incorrectly overwriting the batch information otherwise.